docs/source/howto/data.rst

```diff
@@ -260,6 +260,338 @@ Finding data
 Organizing data
 ===============
 
+Groups
+------
+Group is a tool to organize the nodes of the provenance graph into sub sets.
+Any number of groups can be created and each group can contain any number of nodes of any type.
+
+Typically, you want to put multiple nodes into a group because they share some common property, and through the group they can easily be referenced.
+Unlike nodes, groups can be modified at any time.
+Here we profide a list of typical operations that may be performed with groups:
```

Comment:
I would start here with a quick description of the problem that this section is aiming to address. I imagine this makes it easier for the user to recognize whether they are in the right section, for what they are looking for. Something like:

> AiiDA's database is great for automatically storing all your data, but sometimes it can be tricky to navigate this flat data store.
> To create some order in this mass of data, you can *group* sets of nodes together, just as you would with files in folders on your filesystem.
> A folder, in this analogy, is represented by the :py:class:`~aiida.orm.groups.group.Group` class.
> Each ``Group`` instance can hold any amount of nodes and any node can be contained in any number of nodes.
> A typical use case is to store all nodes that share a common property in a single group.

Before you actually start to add these changes, @csadorf do you agree with this direction/wording?

Reply:
I think it's worthwhile to try to motivate this section by briefly illustrating the underlying problem. 👍

Reply:
changed